From d1c2915742980cb617177291878bd64d5e9651f7 Mon Sep 17 00:00:00 2001
Message-ID: <d1c2915742980cb617177291878bd64d5e9651f7.1753796558.git.lorenzo@kernel.org>
In-Reply-To: <00275548ba8f8ff8da354e44dd25b329e84924f4.1753796558.git.lorenzo@kernel.org>
References: <00275548ba8f8ff8da354e44dd25b329e84924f4.1753796558.git.lorenzo@kernel.org>
From: Lorenzo Bianconi <lorenzo@kernel.org>
Date: Sat, 19 Jul 2025 12:21:10 +0200
Subject: [PATCH 05/22] wifi: mt76: Remove q->ndesc check in mt76_dma_rx_fill()

Remove q->ndesc check in mt76_dma_rx_fill routine since this is already
done in mt76_dma_rx_fill_buf().

Signed-off-by: Lorenzo Bianconi <lorenzo@kernel.org>
---
 dma.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/dma.c b/dma.c
index 2bb6ce3e..5aa33342 100644
--- a/dma.c
+++ b/dma.c
@@ -671,9 +671,6 @@ int mt76_dma_rx_fill(struct mt76_dev *dev, struct mt76_queue *q,
 {
 	int frames;
 
-	if (!q->ndesc)
-		return 0;
-
 	spin_lock_bh(&q->lock);
 	frames = mt76_dma_rx_fill_buf(dev, q, allow_direct);
 	spin_unlock_bh(&q->lock);
-- 
2.50.1

